client/a8c-for-agencies/sections/referrals/referrals-list/subscription-status.tsx

```diff
@@ -10,34 +10,39 @@ export default function SubscriptionStatus( { item }: { item: Referral } ): Reac
 		item: Referral
 	): {
 		status: string | null;
-		type: 'warning' | 'success' | null;
+		type: 'warning' | 'success' | 'info' | null;
 	} => {
 		const activeStatuses = item.statuses.filter( ( status ) => status === 'active' );
 		const pendingStatuses = item.statuses.filter( ( status ) => status === 'pending' );
+		const canceledStatuses = item.statuses.filter( ( status ) => status === 'canceled' );
 
-		if ( ! item.statuses.length ) {
-			return {
-				status: null,
-				type: null,
-			};
+		switch ( item.statuses.length ) {
+			case 0:
+				return {
+					status: null,
+					type: null,
+				};
+			case activeStatuses.length:
+				return {
+					status: translate( 'Active' ),
+					type: 'success',
+				};
+			case pendingStatuses.length:
+				return {
+					status: translate( 'Pending' ),
+					type: 'warning',
+				};
+			case canceledStatuses.length:
+				return {
+					status: translate( 'Canceled' ),
+					type: 'info',
+				};
+			default:
+				return {
+					status: translate( 'Mixed' ),
+					type: 'warning',
+				};
```

Comment:
Instead of looping here trice, we can do it with 1 loop.

```

		if ( ! item.statuses.length ) {
			return {
				status: null,
				type: null,
			};
		}

		const status = item.statuses.reduce( ( prev, curr ) => {
			if ( prev === curr ) {
				return curr;
			}

			return 'mixed';
		}, item.statuses[ 0 ] );

		switch ( status ) {
			case 'active':
				return {
					status: translate( 'Active' ),
					type: 'success',
				};
			case 'pending':
				return {
					status: translate( 'Pending' ),
					type: 'warning',
				};
			case 'canceled':
				return {
					status: translate( 'Canceled' ),
					type: 'info',
				};
			default:
				return {
					status: translate( 'Mixed' ),
					type: 'warning',
				};
		}
```

Reply:
Good suggestion. Fixed it here: https://github.com/Automattic/wp-calypso/pull/91976/commits/91b9859e9c742c28258c89f2f96a035136e6d278